pymongo_auth_aws/auth.py

```diff
@@ -45,6 +46,19 @@ def _aws_temp_credentials():
     if access_key and secret_key:
         return AwsCredential(
             access_key, secret_key, os.environ.get('AWS_SESSION_TOKEN'))
+    # Check if environment variables exposed by IAM Roles for Service Accounts (IRSA) are present.
+    # See https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html for details.
+    irsa_web_id_file = os.getenv('AWS_WEB_IDENTITY_TOKEN_FILE')
+    irsa_role_arn = os.getenv('AWS_ROLE_ARN')
+    if irsa_web_id_file and irsa_role_arn:
+        try:
+            with open(irsa_web_id_file) as f:
+                irsa_web_id_token = f.read()
+            return _irsa_assume_role(irsa_role_arn, irsa_web_id_token, 'pymongo-auth-aws')
```

Comment:
Is `RoleSessionName=pymongo-auth-aws` important for your use case? Are we free to change this string in a future release without warning?

Reply:
Specifically, I'm wondering if we're going to need to standardize this name across all language apis for MongoDB. Perhaps "mongodb-aws" or "MONGODB-AWS" may be a better name?

Reply:
According to https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html I believe this should be `os.getenv('AWS_ROLE_SESSION_NAME')`:

> AWS_ROLE_SESSION_NAME
> Specifies the name to attach to the role session. This value is provided to the RoleSessionName parameter when the AWS CLI calls the AssumeRole operation, and becomes part of the assumed role user ARN: arn:aws:sts::123456789012:assumed-role/role_name/role_session_name. This is an optional parameter. If you do not provide this value, a session name is generated automatically. This name appears in AWS CloudTrail logs for entries associated with this session.
> 
> If defined, this environment variable overrides the value for the profile setting [role_session_name](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-config-role_session_name).
> 
> Used with the AWS_ROLE_ARN and AWS_WEB_IDENTITY_TOKEN_FILE environment variables.
> 
> For more information on using web identities, see [Assume role with web identity](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html#cli-configure-role-oidc).

Reply:
I'm not wedded to the name, but I wouldn't count on that environment variable being present in all deployments. At present, `AWS_ROLE_SESSION_NAME` isn't provided by a default IRSA deployment; from https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-technical-overview.html:

> The [Amazon EKS Pod Identity Webhook](https://github.com/aws/amazon-eks-pod-identity-webhook) on the cluster watches for pods that are associated with service accounts with this annotation and applies the following environment variables to them.
> ```
> AWS_ROLE_ARN=arn:aws:iam::111122223333:role/iam-role-name
>AWS_WEB_IDENTITY_TOKEN_FILE=/var/run/secrets/eks.amazonaws.com/serviceaccount/token
>```

So, would it make sense to check for this env var and default to `pymongo-auth-aws` if not set?

Reply:
Pushed a commit to handle this. If the user cares enough, they can specify the env var to suit their own bookkeeping purposes this way.